app/controllers/profile_controller.rb

```diff
@@ -89,4 +114,10 @@ def mfa_page_erros
     @secret_code_svg = generate_new_qr(secret_code: @secret_code, email: current_user.email)
     render :show_change_mfa, status: :bad_request
   end
+
+  def refresh_user(updated_user)
+    current_user.first_name = updated_user.user_attributes.find { |attr| attr.name == 'given_name' }.value
+    current_user.last_name = updated_user.user_attributes.find { |attr| attr.name == 'family_name' }.value
+    current_user.email = updated_user.user_attributes.find { |attr| attr.name == 'email' }.value
```

Comment:
How come we need to also update the email?

Reply:
So it doesn't directly relate to this story. Currently admins have the ability to change a users email, and that email wont show up as changed on the users profile unless they sign out (i think). And we may later chose to add the ability for users themselves to change their email address so therefore is good to have it in place now.